core/src/test/java/com/linecorp/armeria/server/HttpServerUpgradeHandlerTest.java

```diff
@@ -195,4 +211,31 @@ void upgradeFail() {
         assertThat((Object) channel.readOutbound()).isNull();
         channel.finishAndReleaseAll();
     }
+
+    @Test
+    void upgradeFailWithInvalidSettingsHeader() throws Exception {
```

Comment:
Could we repeat this test twice to make sure we don't have `IllegalRefCntException` or corruption because we didn't duplicate the buffer for each error response? You can use `RepeatedTest` annotation from JUnit.

Reply:
@ahnyujin Could you address @trustin's issue, please? We are about to release a new version, so please let me know if you don't have time for this. We are going to fix that. 😄